src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.AccessibleObject.cs

```diff
@@ -244,7 +244,7 @@ internal override int[] RuntimeId
                     case UiaCore.UIA.HasKeyboardFocusPropertyId:
                         // If no inner cell entire DGV should be announced as focused by Narrator.
                         // Else only inner cell should be announced as focused by Narrator but not entire DGV.
-                        return RowCount == 0;
+                        return _ownerDataGridView.Focused && RowCount == 0;
```

Comment:
It seems, it should be like this:
```suggestion
                        return _ownerDataGridView.Focused;
```
DGV should have `HasKeyboardFocus` when its cell is in focus.
So https://github.com/dotnet/winforms/pull/6102 way was incorrect.
But it should be carefully tested because we had cases when Narrator did an extra focus jump on the whole control.

Reply:
Done, thanks